project/plugins.sbt

```diff
@@ -1,22 +1,16 @@
-addSbtPlugin("com.github.sbt"                    % "sbt-ci-release"                % "1.5.10")
-addSbtPlugin("com.github.cb372"                  % "sbt-explicit-dependencies"     % "0.2.16")
-addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings"              % "3.0.2")
-addSbtPlugin("com.typesafe"                      % "sbt-mima-plugin"               % "1.1.0")
-addSbtPlugin("de.heikoseeberger"                 % "sbt-header"                    % "5.7.0")
-addSbtPlugin("org.portable-scala"                % "sbt-scala-native-crossproject" % "1.2.0")
-addSbtPlugin("org.portable-scala"                % "sbt-scalajs-crossproject"      % "1.2.0")
-addSbtPlugin("org.scala-js"                      % "sbt-scalajs"                   % "1.16.0")
-addSbtPlugin("org.scala-native"                  % "sbt-scala-native"              % "0.4.16")
-addSbtPlugin("org.scalameta"                     % "sbt-mdoc"                      % "2.3.7")
-addSbtPlugin("org.scalameta"                     % "sbt-scalafmt"                  % "2.4.6")
-addSbtPlugin("pl.project13.scala"                % "sbt-jcstress"                  % "0.2.0")
-addSbtPlugin("pl.project13.scala"                % "sbt-jmh"                       % "0.4.4")
-addSbtPlugin("dev.zio"                           % "zio-sbt-website"               % "0.3.1")
-
-libraryDependencies += "org.snakeyaml" % "snakeyaml-engine" % "2.6"
-
-resolvers ++= Resolver.sonatypeOssRepos("public")
+addSbtPlugin("ch.epfl.scala"                     % "sbt-scalafix"              % "0.13.0")
+addSbtPlugin("com.eed3si9n"                      % "sbt-buildinfo"             % "0.12.0")
+addSbtPlugin("com.github.sbt"                    % "sbt-unidoc"                % "0.5.0")
+addSbtPlugin("com.github.sbt"                    % "sbt-ci-release"            % "1.6.1")
+addSbtPlugin("com.github.cb372"                  % "sbt-explicit-dependencies" % "0.3.1")
+addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings"          % "3.0.2")
+addSbtPlugin("com.typesafe"                      % "sbt-mima-plugin"           % "1.1.4")
+addSbtPlugin("de.heikoseeberger"                 % "sbt-header"                % "5.10.0")
+addSbtPlugin("org.scala-js"                      % "sbt-scalajs"               % "1.17.0")
+addSbtPlugin("org.scala-native"                  % "sbt-scala-native"          % "0.5.5")
+addSbtPlugin("org.scalameta"                     % "sbt-mdoc"                  % "2.6.1")
+addSbtPlugin("org.scalameta"                     % "sbt-scalafmt"              % "2.5.2")
+addSbtPlugin("pl.project13.scala"                % "sbt-jcstress"              % "0.2.0")
+addSbtPlugin("pl.project13.scala"                % "sbt-jmh"                   % "0.4.7")
+addSbtPlugin("dev.zio"                           % "zio-sbt-website"           % "0.3.1")
+addSbtPlugin("com.eed3si9n"                      % "sbt-projectmatrix"         % "0.10.0")
```

Comment:
I am not sure about this. I don't think any other zio project is using this so far. And we have zio-sbt to help us with the builds. I'd prefer to stick to the same solution, unless you can make a compelling argument.

Reply:
To my understanding it should also be a default feature in SBT 2.x

Reply:
With the existing build and the command aliases to target the different platforms and scala versions it was quite error prone to make sure all projects were included. Now you do not have to use the `+` or `++` anymore. All projects have a direct projectref in the build.

Reply:
I think the build is much more concise now and it is targeting all the platforms and Scala versions it can support (the existing build was missing some).

Reply:
ref sbt-projectmatrix insource in sbt 2.x: https://eed3si9n.com/sbt-2.0.0-beta